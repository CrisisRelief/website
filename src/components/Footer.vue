<template>
  <footer id="footer" class="footer">
    <nav id="socials" class="d-flex justify-content-around p-2">
      <span v-for="(social, index) in socials" :key="index" class="m-2 h2">
        <a :aria="social.alt" :href="social.link">
          <i :class="social.icon" />
        </a>
      </span>
    </nav>
    <nav id="license">
      <div class="d-flex justify-content-center px-auto">
        <span class="p-2">
          We
          <FontAwesomeIcon icon="heart" />Open
        </span>
        <span class="p-2">
          <a
            href="https://opendatacommons.org/licenses/odbl/1.0/"
            target="_blank"
            rel="noopener noreferrer"
          >ODBL</a>
        </span>
        <span class="p-2">
          <a
            href="https://opendatacommons.org/licenses/dbcl/1.0/"
            target="_blank"
            rel="noopener noreferrer"
          >DBCL</a>
        </span>
      </div>
    </nav>
  </footer>
</template>

<script>
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";

export default {
  components: { FontAwesomeIcon },
  data() {
    return {
      socials: [
        {
          link: "https://twitter.com/thecrisisapp",
          icon: "fab fa-twitter-square",
          alt: "twitter"
        },
        {
          link: "https://www.facebook.com/thecrisisapp/",
          icon: "fab fa-facebook-square",
          alt: "facebook"
        },
        {
          link: "https://www.instagram.com/thecrisisapp",
          icon: "fab fa-instagram",
          alt: "instagram"
        },
        {
          link: "https://www.linkedin.com/showcase/crisisapp/",
          icon: "fab fa-linkedin",
          alt: "linkedin"
        }
      ]
    };
  }
};
</script>

<style scoped>
#socials {
  background: #f5f5f5;
}
</style>
