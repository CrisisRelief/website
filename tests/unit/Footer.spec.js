import { shallowMount } from "@vue/test-utils";
import Footer from "../../src/components/Footer.vue";

const wrapper = shallowMount(Footer);

describe("Footer", () => {
    it("should match last snapshot", () => {
      expect(wrapper.element).toMatchSnapshot();
    });
    it("should render correct social links and icons", () => {
      const socialData = Footer.data().socials;
      const elements = wrapper.findAll('#socials span a');

      socialData.forEach((item, index) => {
        const element = elements.at(index);

        expect(element.attributes().aria).toBe(item.alt);
        expect(element.attributes().href).toBe(item.link);
  
        const icon = element.find("i");

        expect(icon.classes(item.icon)).toBe(true);
        
      });
    });
});