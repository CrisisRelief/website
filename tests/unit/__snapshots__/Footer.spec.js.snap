// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match last snapshot 1`] = `
<footer
  class="footer"
  id="footer"
>
  <nav
    class="d-flex justify-content-around p-2"
    id="socials"
  >
    <span
      class="m-2 h2"
    >
      <a
        aria="twitter"
        href="https://twitter.com/thecrisisapp"
      >
        <i
          class="fab fa-twitter-square"
        />
      </a>
    </span>
    <span
      class="m-2 h2"
    >
      <a
        aria="facebook"
        href="https://www.facebook.com/thecrisisapp/"
      >
        <i
          class="fab fa-facebook-square"
        />
      </a>
    </span>
    <span
      class="m-2 h2"
    >
      <a
        aria="instagram"
        href="https://www.instagram.com/thecrisisapp"
      >
        <i
          class="fab fa-instagram"
        />
      </a>
    </span>
    <span
      class="m-2 h2"
    >
      <a
        aria="linkedin"
        href="https://www.linkedin.com/showcase/crisisapp/"
      >
        <i
          class="fab fa-linkedin"
        />
      </a>
    </span>
  </nav>
   
  <nav
    id="license"
  >
    <div
      class="d-flex justify-content-center px-auto"
    >
      <span
        class="p-2"
      >
        
        We 
        <fontawesomeicon-stub
          icon="heart"
        />
         Open
      
      </span>
       
      <span
        class="p-2"
      >
        <a
          href="https://opendatacommons.org/licenses/odbl/1.0/"
          rel="noopener noreferrer"
          target="_blank"
        >
          ODBL
        </a>
      </span>
       
      <span
        class="p-2"
      >
        <a
          href="https://opendatacommons.org/licenses/dbcl/1.0/"
          rel="noopener noreferrer"
          target="_blank"
        >
          DBCL
        </a>
      </span>
    </div>
  </nav>
</footer>
`;
